consumer/spring-websocket-consumer/src/main/java/org/springframework/cloud/fn/consumer/websocket/trace/InMemoryTraceRepository.java

```diff
@@ -41,35 +44,50 @@ public class InMemoryTraceRepository {
 
 	private final List<Trace> traces = new LinkedList<>();
 
+	private final Lock tracesLock = new ReentrantLock();
+
 	/**
 	 * Flag to say that the repository lists traces in reverse order.
 	 * @param reverse flag value (default true)
 	 */
 	public void setReverse(boolean reverse) {
-		synchronized (this.traces) {
+		this.tracesLock.lock();
```

Comment:
You just removed our conversion on the matter, but still left lock in the setter.
🤷

Reply:
Once our conversation is finished, I was going to remove them. I'll take care of it right away.

Reply:
But that conversation has not been finished.
You just didn't wait for my response.
I am the owner of the project, that the final decision is on me.

Please, don't remove any conversation: it will be helpful in the future when some one would try to figure out what was an origin of this or that code.
Like exactly someone would ask why the `synchronized` as removed here and lock not added.

Reply:
I just thought I would make another commit after our conversation. Of course, I’ll follow your guidelines. I didn’t mean to do anything wrong, sorry if I caused any confusion.